src/service/action-impl.js

```diff
@@ -155,14 +162,41 @@ export class ActionService {
           this.trigger(dev().assertElement(event.target), 'tap', event);
         }
       });
-    } else if (name == 'submit' || name == 'change') {
+    } else if (name == 'submit') {
+      this.root_.addEventListener(name, event => {
+        this.trigger(dev().assertElement(event.target), name, event);
+      });
+    } else if (name == 'change') {
       this.root_.addEventListener(name, event => {
+        this.addChangeDetails_(event);
         this.trigger(dev().assertElement(event.target), name, event);
       });
     }
   }
 
   /**
+   * Given a browser 'change' event, add `details` property containing the
+   * relevant information for the change that generated the initial event.
+   * @param {!Event} event A `change` event.
+   */
+  addChangeDetails_(event) {
+    const detail = {};
+    const target = event.target;
+    if (event.target.tagName.toLowerCase() === 'input') {
+      const inputType = target.getAttribute('type');
+      const fieldsToInclude = WHITELISTED_INPUT_DATA_[inputType];
+      if (fieldsToInclude) {
+        fieldsToInclude.forEach(field => {
+          const value = target[field];
+          const valueAsNumber = parseFloat(value);
+          detail[field] = isNaN(valueAsNumber) ? value : valueAsNumber;
```

Comment:
ugh `isNaN('') == false` so you need to handle the case where value is: `''`, `'   '`, `true`, `false`, `null` separately as both `isNaN` and `Number()` behave weirdly for those values. 

(Also assumption that `value` is always a string is wrong since we are reading property and not necessarily an attribute)